libs/packages/components/src/lib/autocomplete-search/autocomplete-search.component.ts

```diff
@@ -242,7 +242,13 @@ export class SDSAutocompleteSearchComponent implements ControlValueAccessor {
       this.getResults(searchString);
     }
   }
+  isCleanIconVisible(): boolean {
+    if (this.model && this.model.items.length > 0 && !this.model?.items[0][this.configuration.primaryKeyField])
```

Comment:
Would this work if the second item is null? It seems like it's checking just the first item

Reply:
Yes, this issue specific to the Single mode only, because multi mode the text will not stay in the input